libraries/AP_Camera/AP_RunCam.cpp

```diff
@@ -118,13 +118,24 @@ AP_RunCam::Menu AP_RunCam::_menus[RUNCAM_MAX_DEVICE_TYPES] = {
 	{ 6, { 3, 10, 2, 2, 8 }}, // Runcam 2 4K
 };
 
-AP_RunCam::AP_RunCam()
+const char* AP_RunCam::_models[RUNCAM_MAX_DEVICE_TYPES] = {
+    "SplitMicro",
+    "Split",
+    "Split4k",
+    "Hybrid",
+    "Run24k"
+};
+
+AP_RunCam::AP_RunCam(AP_Camera &frontend, AP_Camera_Params &params, uint8_t instance, uint8_t runcam_instance)
+    : AP_Camera_Backend(frontend, params, instance), _runcam_instance(runcam_instance)
 {
     AP_Param::setup_object_defaults(this, var_info);
-    if (_singleton != nullptr) {
-        AP_HAL::panic("AP_RunCam must be singleton");
+    if (_singleton != nullptr && _singleton->_instance == instance) {
+        AP_HAL::panic("AP_RunCam instance must be a singleton %u\n", instance);
```

Comment:
```suggestion
        AP_HAL::panic("AP_RunCam instance must be a singleton %u", instance);
```

if there are places that need to add a line ending then patch those places.

Reply:
I'm not going to patch all uses of panic() in this PR. There are other places where linefeed is added to panic lines and it looks rubbish without it.

Reply:
Ok fixed